protocol/metrics/consumer_relayserver_client.go

```diff
@@ -0,0 +1,196 @@
+			utils.LavaFormatDebug(fmt.Sprintf("CUC: Skipped this iteration - CCC iter:%d.", cuc.sendID))
+			cuc.lock.Unlock()
+		}
+	}
+}
+
+func (cuc *ConsumerRelayserverClient) SetRelayMetrics(relayMetric *RelayMetrics) error {
+	if cuc == nil {
+		return nil
+	}
+
+	request := UpdateMetricsRequest{
+		RecordDate:      relayMetric.Timestamp.Format("20060102"),
+		Hash:            relayMetric.ProjectHash,
+		Chain:           relayMetric.ChainID,
+		ApiType:         relayMetric.APIType,
+		RelaysInc:       1,
+		CuInc:           int(relayMetric.ComputeUnits),
+		LatencyToAdd:    uint64(relayMetric.Latency),
+		LatencyAvgCount: 1,
+	}
+
+	cuc.lock.Lock()
+	cuc.addQueue = append(cuc.addQueue, request)
+	cuc.lock.Unlock()
+
+	return nil
+}
+
+func (cuc *ConsumerRelayserverClient) sendRequests(sendQueue []UpdateMetricsRequest) {
+	if cuc == nil {
+		return
+	}
+
+	utils.LavaFormatDebug("CUC: Starting sendRequests")
+
+	if len(sendQueue) == 0 {
+		utils.LavaFormatDebug("CUC: sendQueue is nil or empty")
+		utils.LavaFormatDebug("CUC: sendQueue", utils.Attribute{Key: "Value", Value: sendQueue}, utils.Attribute{Key: "Length", Value: len(sendQueue)})
+		return
+	}
+
+	aggregatedRequests := cuc.aggregateRequests(sendQueue)
+
+	if len(aggregatedRequests) == 0 {
+		utils.LavaFormatDebug("CUC: No requests after aggregate")
+		utils.LavaFormatDebug("CUC: sendQueue", utils.Attribute{Key: "Value", Value: sendQueue}, utils.Attribute{Key: "Length", Value: len(sendQueue)})
+		utils.LavaFormatDebug("CUC: aggregatedRequests", utils.Attribute{Key: "Value", Value: aggregatedRequests}, utils.Attribute{Key: "Length", Value: len(aggregatedRequests)})
+		return
+	}
+
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+	}
+
+	jsonData, err := json.Marshal(aggregatedRequests)
+	if err != nil {
+		utils.LavaFormatDebug("CUC: Error marshalling sendQueue to JSON")
+		return
+	}
+
+	var resp *http.Response
+	for i := 0; i < 3; i++ {
+		utils.LavaFormatDebug(fmt.Sprintf("CUC: Attempting to post request - Attempt: %d", i+1))
+		resp, err = client.Post(cuc.endPointAddress+"/updateMetrics", "application/json", bytes.NewBuffer(jsonData))
+		if err != nil {
+			utils.LavaFormatDebug(fmt.Sprintf("CUC: Failed to post request - Attempt: %d, Error: %s", i+1, err.Error()))
+			time.Sleep(2 * time.Second)
+		} else {
+			break
+		}
+	}
+
+	if err != nil {
+		utils.LavaFormatDebug("CUC: Failed to send requests after 3 attempts")
+		return
+	}
+
+	defer resp.Body.Close()
+
+	utils.LavaFormatDebug(fmt.Sprintf("CUC: Successfully sent requests - CCC iter:%d.", cuc.sendID))
+}
+
+func (cuc *ConsumerRelayserverClient) aggregateRequests(reqs []UpdateMetricsRequest) []UpdateMetricsRequest {
+	if cuc == nil {
+		return nil
+	}
+
+	// Create a map to hold the aggregated data
+	aggregated := make(map[string]*UpdateMetricsRequest)
+
+	for _, req := range reqs {
+		// Use the combination of RecordDate, Hash, Chain, and ApiType as the key
+		key := req.RecordDate + req.Hash + req.Chain + req.ApiType
```

Comment:
always good practice to have a method generate the key. pass the request and get the key back.

Reply:
what about this?